include/ia2.h

```diff
@@ -384,6 +388,19 @@ static int protect_pages(struct dl_phdr_info *info, size_t size, void *data) {
   // `SegmentInfo` will have its size field set to zero
   size_t extra_seg_info = info->dlpi_phnum;
 
+  // Add the ignored range for the DYNAMIC section. If the library is built with
+  // relro this section should be part of the relro region, so it at least won't
+  // be trivially writable without an mprotect call.
+  for (size_t i = 0; i < info->dlpi_phnum; i++) {
+    Elf64_Phdr phdr = info->dlpi_phdr[i];
+    if (phdr.p_type == PT_DYNAMIC) {
+      struct AddressRange *dynamic_range = &ignored_ranges[NUM_IGNORED_RANGES - 1];
+      dynamic_range->start = (info->dlpi_addr + phdr.p_vaddr) & ~0xFFFUL;
```

Comment:
We need to ensure that rounding the start down/end up doesn't include sections that should be protected. For ia2_shared_data we do this in libia2/padding.ld, but this is more finicky for some existing sections. There's a prototype that works with GNU ld [here](https://github.com/immunant/IA2-Phase2/pull/121/files#diff-9e28fda1df0859baaa4aed0488c407cba6bb5abac419956850641202e4f00766), but IIRC it doesn't work with lld and might not work with relro since it splits the relro sections have to be contiguous (though I agree that we should use relro).